### DISCUSSION TITLE
How do I use an image plugin with nearest sampling?

### DISCUSSION THREAD

**Discussion by vurpo** [bevyengine/bevy#7606] [Q&A] [answered]
I attempted to just add `ImagePlugin::default_nearest()`, however starting the app with this results in:

```
thread 'main' panicked at 'Resource requested by (bevy_ecs::system::system_param::Res<bevy_render::renderer::render_device::RenderDevice>, bevy_ecs::system::system_param::Res<bevy_render::texture::image::DefaultImageSampler>) does not exist: bevy_render::texture::image::DefaultImageSampler', /Users/tilde/.cargo/registry/src/github.com-1ecc6299db9ec823/bevy_ecs-0.9.1/src/system/system_param.rs:412:17
```

This is on macOS.

The previous discussions about this I can find mention `ImageSettings`, which seems to be gone entirely by now. What's the right way to get nearest sampling for images?

**Comment by rparrett** [comment, marked as answer]
You can use the `set` method on `DefaultPlugins` to configure `ImagePlugin`.

https://github.com/bevyengine/bevy/blob/v0.9.1/examples/2d/sprite_sheet.rs#L7-L8

**Reply by vurpo** [reply]
This is exactly what I tried doing, resulting in the error above (sorry if that wasn't clear enough)

**Reply by vurpo** [reply]
Actually that's what I *thought* I was doing. I don't know what the difference is, but previously I tried to create the DefaultPlugins object with the configured ImagePlugin separately *before* adding it to the app, but now that I wrote it inline all on the same line (like the example says) it stopped crashing.

**Reply by rparrett** [reply]
So there are a couple concerns that using `set` on the `DefaultPlugins` `PluginGroup` solves.

- `ImagePlugin` needs `RenderPlugin` to be set up first.
    You were seeing the sort of error you were seeing because you were adding `ImagePlugin` before `DefaultPlugins`
- We can't add a second copy of `ImagePlugin` after `DefaultPlugins` adds its own.
    You would see another sort of error if you had added `ImagePlugin` after `DefaultPlugins`

With `set`, we can tell `DefaultPlugins` to use our configured version of `ImagePlugins` when the plugin group is built, with everything in the right order.
